spec/tbd_tests_spec.rb

```diff
@@ -2637,6 +4196,251 @@
+    fd = OpenStudio::Model::WindowPropertyFrameAndDivider.new(os_model_FD)
+    expect(fd.setFrameWidth(0.030)).to be(true)   # 30mm (narrow) around glazing
+    expect(fd.setFrameConductance(0.500)).to be(true)
+    window_FD = os_model_FD.getSubSurfaceByName(name)
+    expect(window_FD.empty?).to be(false)
+    window_FD = window_FD.get
+    expect(window_FD.allowWindowPropertyFrameAndDivider).to be(true)
+    expect(window_FD.setWindowPropertyFrameAndDivider(fd)).to be(true)
+    width = window_FD.windowPropertyFrameAndDivider.get.frameWidth
+    expect(width).to be_within(0.001).of(0.030)                # good so far ...
+
+    expect(window_FD.netArea).to be_within(0.01).of(5.58)             # 5.89 (?)
+    front_FD = os_model_FD.getSurfaceByName(nom)
+    expect(front_FD.empty?).to be(false)
+    front_FD = front_FD.get
+    expect(front_FD.grossArea).to be_within(0.01).of(110.54)        # this is OK
+    expect(front_FD.netArea).to be_within(0.01).of(95.49)            # 95.17 (?)
+    expect(front_FD.windowToWallRatio()).to be_within(0.01).of(0.101)  # 0.104 ?
+
+    # If one runs a simulation with the exported file below ("os_model_FD.osm"),
+    # EnergyPlus (HTML) will correctly report that the building WWR (gross
+    # window-wall ratio) will have slightly increased from 71% to 72%, due to
+    # the slight increase in area of the "Office Front Wall Window 1" (from
+    # 5.58 m2 to 5.89 m2). The report clearly distinguishes between the revised
+    # glazing area of 5.58 m2 vs a new framing area of 0.31 m2 for this window.
+    # Finally, the parent surface "Office Front Wall" area will also be
+    # correctly reported as 95.17 m2 (vs 95.49 m2). So OpenStudio is correctly
+    # exporting the subsurface and linked Frame & Divider objects to EnergyPlus
+    # (I have not tested triangular windows).
+    #
+    # Unless I'm missing something, there is an obvious discrepency between the
+    # net area ("Office Front Wall") reported by the OpenStudio API vs
+    # EnergyPlus. This may seem minor when looking at the numbers above, but
+    # keep in mind only one glazed subsurface was modified for the comparison.
+    # This difference could easily reach 5% to 10% for models with many windows,
+    # especially for those with narrow aspect ratios (lots of framing).
+    #
+    # The workaround is pretty straightforward - not rocket science. I'll be
+    # implementing/testing something this for TBD a bit later:
+    #
+    #   if frame_and_divider
+    #     heatloss_from_edge = ( edge[:length] + 2*frame_width ) x PSI
+    #   else
+    #     heatloss_from_edge = edge[:length] x PSI
+    #   end
+    #
+    # A similar workaround will be needed for API-reported building WWR.
+    #
+    # ... subsurface.netArea calculation here could be reconsidered (?):
+    # https://github.com/NREL/OpenStudio/blob/
+    # 70a5549c439eda69d6c514a7275254f71f7e3d2b/src/model/Surface.cpp#L1446
+    os_model_FD.save("os_model_FD.osm", true)
```

Comment:
@macumber: I'm hoping these in-line comments are sufficient to explain the API-reported netArea issues when relying on the Frame & Divider objects. If you have a few minutes, could you go over my reasoning/tests?. The workarounds for TBD (corrected edge lengths, netArea) are easy enough - but the netArea reporting could be revised in the API (IMO). Thanks.

Reply:
@macumber : See framedivider.rb for workarounds.